src/Azure.Functions.Cli/Common/KeyVaultReferencesManager.cs

```diff
@@ -45,19 +48,34 @@ private string GetSecretValue(string value)
 
         private ParseSecretResult ParseSecret(string value)
         {
-            try
+            // If the value is null, then we return nothing, as the subsequent call to
+            // UpdateEnvironmentVariables(settings) will log to the user that the setting
+            // is skipped. We check here, because Regex.Match throws when supplied with a
+            // null value.
+            if (value == null)
             {
-                return ParsePrimaryRegexSecret(value) ?? ParseSecondaryRegexSecret(value);
+                return null;
             }
-            catch
+            // Determine if the secret value is attempting to use a key vault reference
+            var keyVaultReferenceMatch = BasicKeyVaultReferenceRegex.Match(value);
+            if (keyVaultReferenceMatch.Success)
             {
-                throw new FormatException($"Key Vault Reference format invalid: {value}");
+                var referenceString = keyVaultReferenceMatch.Groups["ReferenceString"].Value;
+                var result = ParsePrimaryRegexSecret(referenceString) ?? ParseSecondaryRegexSecret(referenceString);
+                // If we detect that a key vault reference was attempted, but did not match any of
+                // the supported formats, we write a warning to the console.
+                if (result == null)
+                {
+                    ColoredConsole.WriteLine(WarningColor($"Unable to parse the Key Vault reference: {value}"));
```

Comment:
I think this should use `key` in the message, not `value`. People often have sensitive information in app setting values that we should avoid logging